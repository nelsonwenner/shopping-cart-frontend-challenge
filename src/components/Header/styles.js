import styled from 'styled-components'

export const Wrapper = styled.header`
  background-color: ${({ theme }) => theme.colors.secondary};
  color: ${({ theme }) => theme.text.secondary};
  width: 100%;
  height: 5rem;
  transition: all 0.3s ease;

  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
`
