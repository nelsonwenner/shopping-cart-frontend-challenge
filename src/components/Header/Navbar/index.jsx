import React from 'react'

import * as S from './styles'

import Container from '@/components/common/Container/styles'
import avatar from '@/assets/avatar/default.png'

const Navbar = () => (
  <S.Wrapper as={Container}>
    <S.Title>Shopping</S.Title>
    <S.User>
      <img alt="avatar" src={avatar} />
      <span>John Doe</span>
    </S.User>
  </S.Wrapper>
)

export default Navbar
