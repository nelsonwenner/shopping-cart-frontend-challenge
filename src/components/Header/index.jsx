import React from 'react'

import * as S from './styles'

import Navbar from './Navbar'

const Header = () => (
  <S.Wrapper>
    <Navbar />
  </S.Wrapper>
)

export default Header
