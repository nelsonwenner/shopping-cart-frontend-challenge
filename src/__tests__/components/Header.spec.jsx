import React from 'react'
import { shallow } from 'enzyme'

import Header from '@/components/Header'

describe('<Header>', () => {
  it('Should render Header component without fail and matches the snapshot', () => {
    const wrapper = shallow(<Header />)
    expect(wrapper).toMatchSnapshot()
  })
})
