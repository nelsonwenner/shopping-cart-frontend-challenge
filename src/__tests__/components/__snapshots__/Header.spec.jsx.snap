// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header> Should render Header component without fail and matches the snapshot 1`] = `
<styled.header>
  <Navbar />
</styled.header>
`;
